test/fixtures/rules/std_rule_cases/L026.yml

```diff
@@ -110,3 +110,46 @@ test_pass_object_referenced_6:
                 table3
                 on table2.y_id = table3.y_id
     ) as cc
+
+test_pass_object_referenced_7:
+  pass_str: |
+    UPDATE my_table
+    SET row_sum = (
+        SELECT COUNT(*) AS row_sum
+        FROM
+            another_table
+        WHERE
+            another_table.id = my_table.id
+    )
+
+test_fail_object_referenced_7:
+  fail_str: |
+    UPDATE my_table
+    SET row_sum = (
+        SELECT COUNT(*) AS row_sum
+        FROM
+            another_table
+        WHERE
+            another_table.id = my_tableeee.id
+    )
+
+test_pass_object_referenced_8:
+  pass_str: |
+    DELETE FROM agent1
+    WHERE EXISTS(
+        SELECT customer.cust_id FROM customer
+        WHERE agent1.agent_code <> customer.agent_code);
+
+test_pass_two_part_reference_8:
+  pass_str: |
+    delete from public.agent1
+    where exists(
+        select customer.cust_id from customer
+        where agent1.agent_code <> customer.agent_code)
+
+test_pass_two_part_reference_9:
+  pass_str: |
+    delete from public.agent1
+    where exists(
+        select customer.cust_id from customer
+        where public.agent1.agent_code <> customer.agent_code)
```

Comment:
So since the code is just looking at the table reference it doesn't flag this example:
```sql
delete from schema1.agent1
where exists(
    select customer.cust_id from customer
    where schema2.agent1.agent_code <> customer.agent_code)
```

`schema2.agent1` is not the same as `schema1.agent1`

I think we may need to check "do the table references match, if so do the schema references match, if so do the database references match?"
and in the event that one is more qualified than the other then allow it as long as the references present in both match to be lenient, like we have in `test_pass_two_part_reference_8` atm (i.e. `public.agent1` and `agent1` can match. Same for `some_db.some_schema.some_table` and `some_schema.some_table` and `some_table`)

Reply:
I'll take another pass at it. This (i.e. robust name resolution, potentially ambiguous) seems like something we may (sooner or later) want to make reusable.